test/e2e/storage/testsuites/base.go

```diff
@@ -232,7 +232,9 @@ func (r *genericVolumeTestResource) cleanupResource(driver TestDriver, pattern t
 	}
 
 	// Cleanup volume for pre-provisioned volume tests
-	DeleteVolume(driver, volType, r.driverTestResource)
+	if r.volume != nil {
+		r.volume.DeleteVolume()
```

Comment:
Should `r.volume` be set to `nil` after this?

Reply:
`r.volume` is still valid and `r` goes out of scope soon after. I personally wouldn't bother.